### ISSUE TITLE
Problems with zipnum indexes

### ISSUE THREAD

**Issue by despens** [ikreymer/pywb#173]
Zipnum indexes don't seem to be working, or there is something non-obvious-to-me I am missing.

I  created a zipnum index using 

`$ python build_local_zipnum.py indexes/ -s 32 -p ../profiles-all.cdxj`

and copied the resulting contents of this `indexes` directory into the `indexes` directory of my collection.

The `indexes` directory of my collection looks like this:

```
$ ls
cdx-00000.gz  cdx-00011.gz  cdx-00022.gz  cluster.summary  part-00010  part-00021
cdx-00001.gz  cdx-00012.gz  cdx-00023.gz  part-00000       part-00011  part-00022
cdx-00002.gz  cdx-00013.gz  cdx-00024.gz  part-00001       part-00012  part-00023
cdx-00003.gz  cdx-00014.gz  cdx-00025.gz  part-00002       part-00013  part-00024
cdx-00004.gz  cdx-00015.gz  cdx-00026.gz  part-00003       part-00014  part-00025
cdx-00005.gz  cdx-00016.gz  cdx-00027.gz  part-00004       part-00015  part-00026
cdx-00006.gz  cdx-00017.gz  cdx-00028.gz  part-00005       part-00016  part-00027
cdx-00007.gz  cdx-00018.gz  cdx-00029.gz  part-00006       part-00017  part-00028
cdx-00008.gz  cdx-00019.gz  cdx-00030.gz  part-00007       part-00018  part-00029
cdx-00009.gz  cdx-00020.gz  cdx-00031.gz  part-00008       part-00019  part-00030
cdx-00010.gz  cdx-00021.gz  cluster.loc   part-00009       part-00020  part-00031
```

This is the content of the file `cluster.loc`:

```
cdx-00000.gz	cdx-00000.gz
cdx-00001.gz	cdx-00001.gz
cdx-00002.gz	cdx-00002.gz
cdx-00003.gz	cdx-00003.gz
cdx-00004.gz	cdx-00004.gz
cdx-00005.gz	cdx-00005.gz
cdx-00006.gz	cdx-00006.gz
cdx-00007.gz	cdx-00007.gz
cdx-00008.gz	cdx-00008.gz
cdx-00009.gz	cdx-00009.gz
cdx-00010.gz	cdx-00010.gz
cdx-00011.gz	cdx-00011.gz
cdx-00012.gz	cdx-00012.gz
cdx-00013.gz	cdx-00013.gz
cdx-00014.gz	cdx-00014.gz
cdx-00015.gz	cdx-00015.gz
cdx-00016.gz	cdx-00016.gz
cdx-00017.gz	cdx-00017.gz
cdx-00018.gz	cdx-00018.gz
cdx-00019.gz	cdx-00019.gz
cdx-00020.gz	cdx-00020.gz
cdx-00021.gz	cdx-00021.gz
cdx-00022.gz	cdx-00022.gz
cdx-00023.gz	cdx-00023.gz
cdx-00024.gz	cdx-00024.gz
cdx-00025.gz	cdx-00025.gz
cdx-00026.gz	cdx-00026.gz
cdx-00027.gz	cdx-00027.gz
cdx-00028.gz	cdx-00028.gz
cdx-00029.gz	cdx-00029.gz
cdx-00030.gz	cdx-00030.gz
cdx-00031.gz	cdx-00031.gz
```

pywb startup shows a warning like the following for ever file name looking like `part-*` or `cdx-*.gz`

`2016-03-22 11:27:07,508: [WARNING]: skipping unrecognized URI: /home/despensr/Projects/staging.webenact/collections/rhizome-profiles/indexes/cdx-00014.gz`

Upon accessing the collection at `http://localhost:8080/rhizome-profiles/*/rhizome.org/profile`, this error message is displayed in the browser:

```pytb
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/pywb/framework/wsgi_wrappers.py", line 62, in handle_methods
    response = wb_router(env)
  File "/usr/local/lib/python2.7/dist-packages/pywb/framework/proxy.py", line 40, in __call__
    response = super(ProxyArchivalRouter, self).__call__(env)
  File "/usr/local/lib/python2.7/dist-packages/pywb/framework/archivalrouter.py", line 43, in __call__
    return route.handler(wbrequest)
  File "/usr/local/lib/python2.7/dist-packages/pywb/webapp/handlers.py", line 87, in __call__
    return self.handle_request(wbrequest)
  File "/usr/local/lib/python2.7/dist-packages/pywb/webapp/handlers.py", line 151, in handle_request
    cdx_lines, output = self.index_reader.load_for_request(wbrequest)
  File "/usr/local/lib/python2.7/dist-packages/pywb/webapp/query_handler.py", line 84, in load_for_request
    cdx_iter = self.load_cdx(wbrequest, params)
  File "/usr/local/lib/python2.7/dist-packages/pywb/webapp/query_handler.py", line 106, in load_cdx
    cdx_iter = self.cdx_server.load_cdx(**params)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxserver.py", line 80, in load_cdx
    return self._check_cdx_iter(cdx_iter, query)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxserver.py", line 45, in _check_cdx_iter
    cdx_iter = self.peek_iter(cdx_iter)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxserver.py", line 88, in peek_iter
    first = next(iterable)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxops.py", line 130, in 
    return (cdx for cdx, _ in itertools.izip(cdx_iter, xrange(limit)))
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxops.py", line 226, in cdx_filter
    for cdx in cdx_iter:
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxops.py", line 120, in 
    return (cls(line) for line in text_iter)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxops.py", line 106, in create_merged_cdx_gen
    for cdx in cdx_iter:
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/zipnum.py", line 151, in gen_cdx
    for blk in blocks:
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/zipnum.py", line 287, in idx_to_cdx
    yield self.block_to_cdx_iter(blocks, ranges, query)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/zipnum.py", line 300, in block_to_cdx_iter
    return self.load_blocks(location, blocks, ranges, query)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/zipnum.py", line 321, in load_blocks
    reader = self.blk_loader.load(location, blocks.offset, blocks.length)
  File "/usr/local/lib/python2.7/dist-packages/pywb/utils/loaders.py", line 162, in load
    return loader.load(url, offset, length)
  File "/usr/local/lib/python2.7/dist-packages/pywb/utils/loaders.py", line 217, in load
    afile = open(url, 'rb')
IOError: [Errno 2] No such file or directory: 'cdx-00001.gz'
```

If I am changing `cluster.loc` to contain a path, like this:

```
cdx-00000.gz	indexes/cdx-00000.gz
cdx-00001.gz	indexes/cdx-00001.gz
cdx-00002.gz	indexes/cdx-00002.gz
cdx-00003.gz	indexes/cdx-00003.gz
cdx-00004.gz	indexes/cdx-00004.gz
cdx-00005.gz	indexes/cdx-00005.gz
….
```
the following error is shown ("No module named indexes"):

```pytb
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/pywb/framework/wsgi_wrappers.py", line 62, in handle_methods
    response = wb_router(env)
  File "/usr/local/lib/python2.7/dist-packages/pywb/framework/proxy.py", line 40, in __call__
    response = super(ProxyArchivalRouter, self).__call__(env)
  File "/usr/local/lib/python2.7/dist-packages/pywb/framework/archivalrouter.py", line 43, in __call__
    return route.handler(wbrequest)
  File "/usr/local/lib/python2.7/dist-packages/pywb/webapp/handlers.py", line 87, in __call__
    return self.handle_request(wbrequest)
  File "/usr/local/lib/python2.7/dist-packages/pywb/webapp/handlers.py", line 151, in handle_request
    cdx_lines, output = self.index_reader.load_for_request(wbrequest)
  File "/usr/local/lib/python2.7/dist-packages/pywb/webapp/query_handler.py", line 84, in load_for_request
    cdx_iter = self.load_cdx(wbrequest, params)
  File "/usr/local/lib/python2.7/dist-packages/pywb/webapp/query_handler.py", line 106, in load_cdx
    cdx_iter = self.cdx_server.load_cdx(**params)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxserver.py", line 80, in load_cdx
    return self._check_cdx_iter(cdx_iter, query)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxserver.py", line 45, in _check_cdx_iter
    cdx_iter = self.peek_iter(cdx_iter)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxserver.py", line 88, in peek_iter
    first = next(iterable)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxops.py", line 130, in 
    return (cdx for cdx, _ in itertools.izip(cdx_iter, xrange(limit)))
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxops.py", line 226, in cdx_filter
    for cdx in cdx_iter:
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxops.py", line 120, in 
    return (cls(line) for line in text_iter)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/cdxops.py", line 106, in create_merged_cdx_gen
    for cdx in cdx_iter:
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/zipnum.py", line 151, in gen_cdx
    for blk in blocks:
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/zipnum.py", line 287, in idx_to_cdx
    yield self.block_to_cdx_iter(blocks, ranges, query)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/zipnum.py", line 300, in block_to_cdx_iter
    return self.load_blocks(location, blocks, ranges, query)
  File "/usr/local/lib/python2.7/dist-packages/pywb/cdx/zipnum.py", line 321, in load_blocks
    reader = self.blk_loader.load(location, blocks.offset, blocks.length)
  File "/usr/local/lib/python2.7/dist-packages/pywb/utils/loaders.py", line 162, in load
    return loader.load(url, offset, length)
  File "/usr/local/lib/python2.7/dist-packages/pywb/utils/loaders.py", line 229, in load
    pkg_split[1])
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1158, in resource_stream
    return get_provider(package_or_requirement).get_resource_stream(
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 427, in get_provider
    __import__(moduleOrReq)
ImportError: No module named indexes
```

**Comment by ikreymer** [comment]
This is caused by the fact that the paths in `cluster.loc` are relative to the working dir, not relative to the collection.. It should be fixed so that, unless a scheme is specified, the paths are resolved relative to `cluster.loc`. Additionally, the `build_local_zipnum.py` script should not use the subdirectory unless it places shards in the appropriate subdir as well.

A temporary workaround is to use an alternative to `cluster.loc`, specifying the path implicitly with this config

```
shard_index_loc:
    match: '(collections/[^/]+/)'
    replace: '\1'
```

**Comment by ikreymer** [comment]
Fixed in upcoming python 3 branch (0.30.0) and also backported to 0.11.5
